digitalpy/core/component_management/controllers/component_management_general_controller.py

```diff
@@ -1,57 +1,136 @@
 #######################################################
-# 
+#
 # core_name_general_controller.py
 # Python implementation of the Class CoreNameGeneralController
 # Generated by Enterprise Architect
 # Created on:      16-Dec-2022 10:56:05 AM
 # Original author: Giu Platania
-# 
+#
 #######################################################
-from Catalog.Implementation.Libraries.Digitalpy.digitalpy.Async.routing.controller import Controller
+from typing import Any, Dict, List
+from digitalpy.core.main.controller import Controller
+
 
 class ComponentManagementGeneralController(Controller):
-# default constructor  def __init__(self):  
+    def __init__(self, component_registry, **kwargs):
+        super().__init__(**kwargs)
+        self.component_registry = component_registry
+
+    def get_component_info(self, component_id: str) -> Dict:
```

Comment:
**[pep8]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>D102 Missing docstring in public method